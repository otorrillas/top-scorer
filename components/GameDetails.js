import React from 'react';
import PropTypes from 'prop-types';

const GameDetails = ({ time, video, player }) => {
  const minutes = Math.floor(time / 60);
  const seconds = time % 60;

  return (
    <>
      <p>
        <b>Time:</b> {minutes} minutes and {seconds} seconds
      </p>
      <p>
        <b>Player:</b> {player}
      </p>
      <button>
        <a href={video} target="_blank" rel="noopener noreferrer">
          Watch run video
        </a>
      </button>
    </>
  );
};

GameDetails.propTypes = {
  time: PropTypes.number.isRequired,
  video: PropTypes.string.isRequired,
  player: PropTypes.string.isRequired
};

export default GameDetails;
