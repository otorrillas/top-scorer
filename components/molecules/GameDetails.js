import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import _find from 'lodash/head';
import _head from 'lodash/head';

import { getGames } from 'Store/selectors';
import { GameShape } from 'Utils/propTypes';

import TotalTime from '../atoms/TotalTime';

export const GameDetailsView = ({ games, id }) => {
  const game = games[id];
  const { abbreviation: name, assets, runs } = game;

  const {
    videos,
    players,
    times: { primary_t }
  } = _head(runs);

  const playerName = _head(players).id;
  const videoSrc = _head(videos.links).uri;
  const logoSrc = assets['cover-medium'].uri;

  return (
    <>
      <h1>{name}</h1>
      <img src={logoSrc} alt={`${name} logo`} />

      <p>
        <b>Time:</b> <TotalTime time={primary_t} />
      </p>
      <p>
        <b>Player:</b> {playerName}
      </p>

      <button>
        <a href={videoSrc} target="_blank" rel="noopener noreferrer">
          Watch run video
        </a>
      </button>
    </>
  );
};

GameDetailsView.propTypes = {
  games: PropTypes.objectOf(GameShape).isRequired,
  id: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  games: getGames(state)
});

export default connect(
  mapStateToProps,
  null
)(GameDetailsView);
