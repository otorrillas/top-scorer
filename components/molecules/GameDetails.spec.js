import React from 'react';
import renderer from 'react-test-renderer';

import { GameDetailsView } from './GameDetails';

export const mockGames = {
  '1234': {
    abbreviation: 'test',
    assets: {
      'cover-medium': {
        uri: 'https://cover-medium-url'
      }
    },
    runs: [
      {
        videos: { links: [{ uri: 'https://video-url' }] },
        players: [{ id: 'playerId' }],
        times: { primary_t: 450 }
      }
    ]
  }
};

describe('<GameDetails />', () => {
  const renderComponent = props =>
    renderer.create(<GameDetailsView {...props} />);

  describe('@render', () => {
    it('default', () => {
      const tree = renderComponent({
        games: mockGames,
        id: '1234'
      });
      expect(tree).toMatchSnapshot();
    });
  });
});
