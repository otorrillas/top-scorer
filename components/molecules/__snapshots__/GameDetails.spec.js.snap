// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameDetails /> @render default 1`] = `
Array [
  <h1>
    test
  </h1>,
  <img
    alt="test logo"
    src="https://cover-medium-url"
  />,
  <p>
    <b>
      Time:
    </b>
     
    7
     minutes and 
    30
     seconds
  </p>,
  <p>
    <b>
      Player:
    </b>
     
    playerId
  </p>,
  <button>
    <a
      href="https://video-url"
      rel="noopener noreferrer"
      target="_blank"
    >
      Watch run video
    </a>
  </button>,
]
`;
